MetaMask/metamask-extension
# fix: UX: NetworkList: Ensure setNetworkClientIdForDomain is called for test networks

## **Description**

This PR removes duplicate code for NetworkListItem creation, instead using a helper function to create a single item.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/25195?quickstart=1)

## **Related issues**

Fixes: N/A

## **Manual testing steps**

1. Click "Polygon" and see chain switch to Polygon
2. Click "Ethereum Mainnet" and see chain switch to Ethereum Mainnet
3. Drag and drop networks around -- should work
4. Click the toggle to view test networks
5. Click a test network -- should change to correct network

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
